server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -178,165 +111,117 @@ export class MetadataExtractionProcessor {
-      if (stream.codec_type === 'video') {
-        newExif.exifImageWidth = stream.width || null;
-        newExif.exifImageHeight = stream.height || null;
-
-        if (typeof stream.rotation === 'string') {
-          newExif.orientation = stream.rotation;
-        } else if (typeof stream.rotation === 'number') {
-          newExif.orientation = `${stream.rotation}`;
-        } else {
-          newExif.orientation = null;
-        }
-
-        if (stream.r_frame_rate) {
-          const fpsParts = stream.r_frame_rate.split('/');
-
-          if (fpsParts.length === 2) {
-            newExif.fps = Math.round(parseInt(fpsParts[0]) / parseInt(fpsParts[1]));
-          }
-        }
-      }
-    }
-
-    await this.exifRepository.upsert(newExif, { conflictPaths: ['assetId'] });
-    await this.assetRepository.save({ id: asset.id, duration: durationString, fileCreatedAt });
+    await this.exifRepository.upsert(exifData, { conflictPaths: ['assetId'] });
+    await this.assetRepository.save({
+      id: asset.id,
+      duration: Duration.fromObject({ seconds: tags.Duration }).toFormat('hh:mm:ss.SSS'),
+      fileCreatedAt: exifData.dateTimeOriginal ?? undefined,
+    });
 
     return true;
   }
 
-  private async applyReverseGeocoding(asset: AssetEntity, newExif: ExifEntity) {
-    const { latitude, longitude } = newExif;
-    if (this.reverseGeocodingEnabled && longitude && latitude) {
-      try {
-        const { country, state, city } = await this.geocodingRepository.reverseGeocode({ latitude, longitude });
-        newExif.country = country;
-        newExif.state = state;
-        newExif.city = city;
-      } catch (error: any) {
-        this.logger.warn(
-          `Unable to run reverse geocoding due to ${error} for asset ${asset.id} at ${asset.originalPath}`,
-          error?.stack,
-        );
-      }
+  private async applyReverseGeocoding(asset: AssetEntity, exifData: ExifEntity): Promise<ExifEntity> {
+    const { latitude, longitude } = exifData;
+    if (!this.reverseGeocodingEnabled || !longitude || !latitude) {
+      return exifData;
     }
-  }
 
-  private extractDuration(duration: number | string | null) {
-    const videoDurationInSecond = Number(duration);
-    if (!videoDurationInSecond) {
-      return null;
+    try {
+      const geocode = await this.geocodingRepository.reverseGeocode({ latitude, longitude });
+      return { ...exifData, ...geocode };
+    } catch (error: any) {
+      this.logger.warn(
+        `Unable to run reverse geocoding due to ${error} for asset ${asset.id} at ${asset.originalPath}`,
+        error?.stack,
+      );
     }
+    return exifData;
+  }
+
+  private async exifData(asset: AssetEntity): Promise<[ExifEntity, ImmichTags]> {
+    const readTaskOptions = {
+      defaultVideosToUTC: false,
+      useMWG: true,
+      numericTags: ['Duration', 'FocalLength'],
+      includeImageDataMD5: false,
+      geoTz: (lat: number, lon: number): string => geotz.find(lat, lon)[0],
```

Comment:
This package is recommended for use by the `exiftool-vendored` authors as it's more accurate, though "larger" and slower than the default. We've discussed this and concluded we'd prefer the accuracy as the performance impact should be negligible.